File: src/main/java/de/fluxparticle/animation/object/ElementPath.java
package de.fluxparticle.animation.object;

import de.fluxparticle.animation.point.DynamicPoint2D;
import javafx.beans.binding.DoubleExpression;

public class ElementPath extends ElementNode {

	private final DynamicPoint2D origin;

	private final DynamicPoint2D dest;

	public ElementPath(DoubleExpression time, DynamicPoint2D origin, DynamicPoint2D dest) {
		super(time);
		this.origin = origin;
		this.dest = dest;
	}

	public DynamicPoint2D getOrigin() {
		return origin;
	}

	public DynamicPoint2D getDest() {
		return dest;
	}

	@Override
	public <R> R accept(ElementNodeVisitor<R> visitor) {
		return visitor.visitPath(this);
	}

}

File: src/main/kotlin/de/fluxparticle/animation/object/ElementPath.kt
package de.fluxparticle.animation.`object`

import de.fluxparticle.animation.point.DynamicPoint2D
import javafx.beans.binding.DoubleExpression

class ElementPath(time: DoubleExpression, val origin: DynamicPoint2D, val dest: DynamicPoint2D) : ElementNode(time) {

    override fun <R> accept(visitor: ElementNodeVisitor<R>): R {
        return visitor.visitPath(this)
    }

}
